crates/nu-parser/src/parser.rs

```diff
@@ -3137,24 +3149,25 @@ pub fn parse_var_with_opt_type(
         let var_name = bytes;
 
         if !is_variable(&var_name) {
-            working_set.error(ParseError::Expected(
-                "valid variable name",
-                spans[*spans_idx],
-            ));
-            return (garbage(spans[*spans_idx]), None);
+            working_set.error(ParseError::Expected("valid variable name", spans.current()));
+            return (garbage(spans.current()), None);
         }
+        // let next_span = spans.peek_next() else {
+        //     working_set.error(ParseError::Expected("valid variable name", spans.current()));
+        //     return (garbage(spans.current()), None);
+        // };
 
-        let id = working_set.add_variable(
-            var_name,
-            span(&spans[*spans_idx..*spans_idx + 1]),
-            Type::Any,
-            mutable,
-        );
+        // Question: Why was this
+        //    span(&spans[*spans_idx..*spans_idx + 1]),
+        // instead of the equivalent
+        //    spans[*spans_idx]
+
+        let id = working_set.add_variable(var_name, spans.current(), Type::Any, mutable);
 
         (
             Expression {
                 expr: Expr::VarDecl(id),
-                span: span(&spans[*spans_idx..*spans_idx + 1]),
+                span: spans.current(),
```

Comment:
why were we recreating a single span in a complicated way here?

```
span(&spans[*spans_idx..*spans_idx + 1]),
```
is equivalent to
```
spans[*spans_idx]
```
as far as I can tell